src/Diagnostics.RuntimeHost/Services/SearchService/SearchService.cs

```diff
@@ -45,39 +49,47 @@ public Task<HttpResponseMessage> Get(HttpRequestMessage request)
             return _httpClient.SendAsync(request);
         }
 
-        public Task<HttpResponseMessage> SearchDetectors(string requestId, string query, Dictionary<string, string> parameters)
+        public async Task<HttpResponseMessage> SearchDetectors(string requestId, string query, Dictionary<string, string> parameters)
         {
             parameters.Add("text", query);
             parameters.Add("requestId", requestId ?? string.Empty);
             HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Post, QueryDetectorsUrl);
+            string authToken = await SearchServiceTokenService.Instance.GetAuthorizationTokenAsync();
+            request.Headers.Add("Authorization", authToken);
             request.Content = new StringContent(JsonConvert.SerializeObject(parameters), Encoding.UTF8, "application/json");
-            return Get(request);
+            return await Get(request);
         }
 
-        public Task<HttpResponseMessage> SearchUtterances(string requestId, string query, string[] detectorUtterances, Dictionary<string, string> parameters)
+        public async Task<HttpResponseMessage> SearchUtterances(string requestId, string query, string[] detectorUtterances, Dictionary<string, string> parameters)
         {
             parameters.Add("detector_description", query);
             parameters.Add("detector_utterances", JsonConvert.SerializeObject(detectorUtterances));
             parameters.Add("requestId", requestId);
             HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Post, QueryUtterancesUrl);
+            string authToken = await SearchServiceTokenService.Instance.GetAuthorizationTokenAsync();
```

Comment:
You could reduce copied code between these methods by having a helper function to add the `requestId`, `Authorization`, and optional `Content` to the request.